libs/dal/src/repositories/message/message.repository.ts

```diff
@@ -430,6 +439,122 @@ export class MessageRepository extends BaseRepository<MessageDBModel, MessageEnt
     );
   }
 
+  /**
+   * Allows to update the status of multiple messages at once.
+   * The status can be updated to seen, unseen, read, unread, archived or unarchived.
+   * Depending on the flag passed, the other flags will be updated accordingly.
+   * For example:
+   * seen -> { seen: true }
+   * read -> { seen: true, read: true }
+   * archived -> { seen: true, read: true, archived: true }
+   * unseen -> { seen: false, read: false, archived: false }
+   * unread -> { seen: true, read: false, archived: false }
+   * unarchived -> { seen: true, read: true, archived: false }
+   */
+  async updateMessagesStatus({
```

Comment:
before checking the implementation please read the doc comment above

Reply:
Nice. I think this section deserves a comment.